docs/inline_shortcodes.md

```diff
@@ -0,0 +1,119 @@
+
+- [Converting Wordpress inline shortcodes for Draftail editor](#converting-wordpress-inline-shortcodes-for-draftail-editor)
+  - [Shortcode handler class](#shortcode-handler-class)
+  - [How to create your own inline shortcode handler](#how-to-create-your-own-inline-shortcode-handler)
+  - [An example of how to add a shortcode handler for the Draftail editor](#an-example-of-how-to-add-a-shortcode-handler-for-the-draftail-editor)
+    - [Stock inline shortcode handler class](#stock-inline-shortcode-handler-class)
+    - [Stock inline shortcode handler configuration](#stock-inline-shortcode-handler-configuration)
+  - [How to test this example works](#how-to-test-this-example-works)
+
+## Shortcode handler class
+
+The package is able to transform Wordpress inline shortcodes.
+
+We provide a base class [InlineShortcodeHandler](/wagtail-wordpress-import/wagtail_wordpress_import/handle_inline_shortcodes.py) that performs regular expression search for a shortcode.
+
+## How to create your own inline shortcode handler
+
+1. You will need to create a subclass of InlineShortcodeHandler.
+2. Then you will need to extend the Draftail editor to recognise the new HTML tag that your shortcode handler will generate.
+
+## An example of how to add a shortcode handler for the Draftail editor
+
+Extending the Draftail editor is fully documented in the [Wagtail Documentation](https://docs.wagtail.io/en/stable/extending/extending_draftail.html).
+
+This example will enable the stock chooser display in the Draftail editor. It's a fully functioning but somewhat fictional example but should help you get started with your own inline shortcode handler.
+
+The complete Wagtail Documentation example can be [Viewed Here](https://docs.wagtail.io/en/stable/extending/extending_draftail.html#creating-new-entities) and this is the first step you should take to get the example working in your own project.
+
+We are making the assumption that in your imported HTML that will be part of a RichText block there is an inline shortcode that will represent a stock to be displayed both in the editor and in the front end.
+
+The word press shortcode will be transformed during the import form  `[stock symbol="TSLA"]` to `<span data-stock="TSLA">$TSLA</span>` before being saved to the RichText block.
+
+You will only need to write one class to handle a shortcode like this. the symbol is dynamic and can be different across multiple shortcode in hte same HTML content.
+
+All shortcodes of the type `[stock symbol="any-valid-symbol"]` will be transformed.
+
+### Stock inline shortcode handler class
+
+```python
+# you can give this module any name or add the class to an existing module
+
+# import the base handler class
+from wagtail_wordpress_import.handle_inline_shortcodes import (
+    InlineShortcodeHandler, 
+)
+
+
+# Subclasses should declare a shortcode_name and provide
+# a construct_html_tag method for converting the shortcode to a HTML tag.
+# @register()
```

Comment:
**Question**: Should this be uncommented?

Reply:
Ah no @register() is not required for Inline Shortcodes. I will remove that part of the comment.

It's implemented here:

```python
def build_none_block_content(html, blocks):
    """
    image_linker is called to link up and retrive the remote image
    document_linker is called to link up and retrive the remote documents
    filters are called to replace inline shortcodes
    """
    html = image_linker(html)
    html = document_linker(html)
    for inline_shortcode_handler in getattr(
        settings, "WAGTAIL_WORDPRESS_IMPORTER_INLINE_SHORTCODE_HANDLERS", []
    ):
        function = import_string(inline_shortcode_handler).construct_html_tag
        html = function(html)
    blocks.append({"type": "rich_text", "value": html})
    html = ""
    return html
```

Unlike the block shortcodes there no other pre-filters to run at the same place in the import.